app/src/main/java/com/example/concordia_campus_guide/Models/Routes/Bus.java

```diff
@@ -0,0 +1,22 @@
+package com.example.concordia_campus_guide.Models.Routes;
+
+import com.example.concordia_campus_guide.GoogleMapsServicesTools.GoogleMapsServicesModels.DirectionsStep;
+
+public class Bus extends TransportType {
+
+    private String busNumber;
+
+    public Bus() {}
+
+    public Bus(DirectionsStep step) {
+        busNumber = step.transitDetails.line.shortName;
```

Comment:
We should add a null check in case the line/shortName does not exist

Reply:
then we should also do it for every other transport type?

Reply:
we can't assume that there will always be a shortname? Like I know it's a bus and it must have transitdetails, a line and a shortname. Can't I expect google maps api will return me non empty objects for these?

Reply:
I don't think we should assume anything